### PR TITLE
misc: add git3po scripts

### LINKED ISSUE #10192: keep git3po configs in this repo
would be nice :)

### PR DESCRIPTION
**Pull request by paulirish**
fixes #10192 

------------

old approach.. 

i have many lines of this in a every-five-minutes.sh that's run via cron:

```sh
env GIT3PO_GH_TOKEN=<token> node bin/git3po.js -c lighthouserules/rule1.yaml
````

new approach after this lands..

replace those lines with this:

```sh
npm install -g git3po

cd ~/code/lighthouse/lighthouse-core/scripts/git3po-rules

for f in $(ls); do 
  env GIT3PO_GH_TOKEN=<token> git3po -c $f 
done
```

does that sound right @connorjclark @patrickhulce ?

### CODE REVIEW

**Comment by connorjclark** [comment]
can you add a bash file to run these too?

**Comment by paulirish** [comment]
> can you add a bash file to run these too?

good call done.

File: lighthouse-core/scripts/run-git3po.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+set -ex
+
+DIRNAME="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# git3po assumes relative config paths from cwd.
+cd "$DIRNAME"
+
+npm install -g git3po
+
+find git3po-rules/*.yaml -exec git3po -c {} \;
```
Comment (connorjclark) [inline lighthouse-core/scripts/run-git3po.sh]: love it

**Comment by connorjclark** [comment]
writing back to yaml seems fine I guess. but a `--start-at` config option (combined with `date`) is also good enough IMO. Would allow us to omit `startAt` from the config, do an initial first-pass with an early date for new conifgs, and use "1 hour ago" or w/e for the cron job. No need to persist.

File: lighthouse-core/scripts/git3po-rules/cannot-attach-to-this-target.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      text:
+        $or:
+          - $match: 'Extension Error: Cannot attach to this target'
+      state: open
+  - type: comments
+    length:
+      $lt: 2
+actions:
+  - type: add_comment
+    body: 'Howdy chief! Appreciate you filing this bug. :clap:
```
Comment (connorjclark) [inline lighthouse-core/scripts/git3po-rules/cannot-attach-to-this-target.yaml]:
```suggestion
    body: 'Howdy! Appreciate you filing this bug. :clap:
```

I think we should remove "chief" from all of these.

File: lighthouse-core/scripts/git3po-rules/lock-resolved.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+repo: GoogleChrome/lighthouse
```
Comment (connorjclark) [inline lighthouse-core/scripts/git3po-rules/lock-resolved.yaml]: FYI I added this.

File: lighthouse-core/scripts/git3po-rules/land-when-green.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+repo: GoogleChrome/lighthouse
```
Comment (connorjclark) [inline lighthouse-core/scripts/git3po-rules/land-when-green.yaml]: i added this

File: lighthouse-core/scripts/run-git3po.sh (hunk 1 above)
Comment (connorjclark) [reply lighthouse-core/scripts/run-git3po.sh]: we can now remove `startAt` from all the configs and do
```
--start-at=`date "+%s" -d "1 hour ago"`
```

although the autolock one should be special-cased to `1 month ago`

**Comment by connorjclark** [comment]
`remove-needs-priority.yaml`

```yaml
repo: GoogleChrome/lighthouse
filters:
  - type: issue
    criteria:
      state: open
      labels:
        $and:
          - $includes: needs-priority
          - $or:
            - $includes: P0
            - $includes: P1
            - $includes: P1.5
            - $includes: P2
            - $includes: P3
actions:
  - type: remove_label
    label: needs-priority
```

File: lighthouse-core/scripts/git3po-rules/cannot-attach-to-this-target.yaml (hunk 2 above)
Comment (paulirish) [reply lighthouse-core/scripts/git3po-rules/cannot-attach-to-this-target.yaml]: i <3 "chief"

File: lighthouse-core/scripts/run-git3po.sh (hunk 1 above)
Comment (paulirish) [reply lighthouse-core/scripts/run-git3po.sh]: k we do this in a followup.

**Comment by paulirish** [comment]
i reverted your additions. lets add them back once merged. that way we can verify this whole new setup works off the latest stuff.

**Comment by paulirish** [comment]
ok it's setup on my side. 👍 

cronjob runs every 5 minutes. it updates a checkout of lighthouse to master and runs `run-git3po.sh`

**Pull request merged**